build-script/code-serving-doc-parser/src/common/msg_queue.py

```
@@ -0,0 +1,152 @@
+                break
+
+            except Exception as e:
+                self._logger.info(f"[{i + 1}/20] Broker Connection Failed. \n\n{repr(e)}\n\n Retrying...")
+                time.sleep(5)
+                continue
+        else:
+            self._logger.error('Failed to Connect!')
+
+        return
+
+    def __exit__(self):
```

Ruff (0.15.18)

[error] The special method `__exit__` expects 4 parameters, 1 was given

(PLE0302)